extensions/blocks/opentable/edit.js

```diff
@@ -0,0 +1,293 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+import { isEmpty, isEqual, join } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { BlockControls, BlockIcon, InspectorControls } from '@wordpress/block-editor';
+import {
+	Button,
+	ExternalLink,
+	Notice,
+	PanelBody,
+	Placeholder,
+	SelectControl,
+	TextareaControl,
+	ToggleControl,
+	Toolbar,
+} from '@wordpress/components';
+import { useState } from '@wordpress/element';
+import { _x, __ } from '@wordpress/i18n';
+import { ENTER, SPACE } from '@wordpress/keycodes';
+
+/**
+ * Internal dependencies
+ */
+import './editor.scss';
+import icon from './icon';
+import RestaurantPicker from './restaurant-picker';
+
+import {
+	getStyleOptions,
+	getStyleValues,
+	languageOptions,
+	languageValues,
+	defaultAttributes,
+	getValidatedAttributes,
+} from './attributes';
+
+const EmbedCodeForm = ( { onSubmit } ) => {
+	const [ embedCode, setEmbedCode ] = useState( '' );
+
+	return (
+		<form onSubmit={ () => onSubmit( embedCode ) }>
+			<TextareaControl
+				onChange={ value => setEmbedCode( value ) }
+				placeholder={ __( 'Paste your OpenTable embed code here…', 'jetpack' ) }
+			>
+				{ embedCode }
+			</TextareaControl>
+			<div>
+				<Button isLarge type="submit">
+					{ _x( 'Embed', 'button label', 'jetpack' ) }
+				</Button>
+			</div>
+		</form>
+	);
+};
+
+export default function OpentableEdit( { attributes, setAttributes, className, clientId } ) {
+	const validatedAttributes = getValidatedAttributes( defaultAttributes, attributes );
+
+	if ( ! isEqual( validatedAttributes, attributes ) ) {
+		setAttributes( validatedAttributes );
+	}
+
+	const { rid, style, iframe, domain, lang, newtab } = attributes;
+	const [ notice, setNotice ] = useState();
+
+	const setErrorNotice = () =>
+		setNotice(
+			<>
+				<strong>{ __( 'We ran into an issue', 'jetpack' ) }</strong>
+				<br />
+				{ __( 'Please ensure this embed matches the one from your OpenTable account', 'jetpack' ) }
+			</>
+		);
+
+	const parseEmbedCode = embedCode => {
+		if ( ! embedCode ) {
+			setErrorNotice();
+			return;
+		}
+
+		const scriptTagAttributes = embedCode.match( /< *script[^>]*src *= *["']?([^"']*)/i );
+		if ( ! scriptTagAttributes || ! scriptTagAttributes[ 1 ] ) {
+			setErrorNotice();
+			return;
+		}
+
+		let src = '';
+		if ( scriptTagAttributes[ 1 ].indexOf( 'http' ) === 0 ) {
+			src = new URL( scriptTagAttributes[ 1 ] );
+		} else {
+			src = new URL( 'http:' + scriptTagAttributes[ 1 ] );
+		}
+
+		if ( ! src.search ) {
+			setErrorNotice();
+			return;
+		}
+
+		const searchParams = new URLSearchParams( src.search );
```

Comment:
I can't seem to find the JP browsers support list, but just in case [`URLSearchParams` is not compatible with IE11](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams#Browser_compatibility).

Reply:
Ah shame. Good catch

Reply:
Ah shame. Good catch

Reply:
What's our policy on [using polyfills](https://www.npmjs.com/package/url-search-params-polyfill)?

Reply:
Also for future reference, I found some info about browser support at PCYsg-2bz-p2. It's pretty confusing when it comes to IE, but it seems like IE 11 is supported for now because Microsoft still provides security updates etc for current Windows versions.

Reply:
> What's our policy on [using polyfills](https://www.npmjs.com/package/url-search-params-polyfill)?

I'm not sure, but I've found a polyfill for `URL` and `URLSearchParams` used somewhere else in JP, and just imported here as well for good measure: https://github.com/Automattic/jetpack/pull/14220/commits/20b594f0c6d68fb354725ccc7718162dc722e1fe

Reply:
> IE 11 is supported for now

It should be, yes, although things are currently broken :( #14273 

In any case, using a polyfill does seem like a good idea indeed.